Tests/SwiftIfConfigTest/VisitorTests.swift

```diff
@@ -0,0 +1,255 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftDiagnostics
+import SwiftIfConfig
+import SwiftParser
+import SwiftSyntax
+import SwiftSyntaxMacrosGenericTestSupport
+import XCTest
+
+/// Visitor that ensures that all of the nodes we visit are active.
+///
+/// This cross-checks the visitor itself with the `SyntaxProtocol.isActive(in:)`
+/// API.
+class AllActiveVisitor: ActiveSyntaxAnyVisitor<TestingBuildConfiguration> {
+  init(configuration: TestingBuildConfiguration) {
+    super.init(viewMode: .sourceAccurate, configuration: configuration)
+  }
+  open override func visitAny(_ node: Syntax) -> SyntaxVisitorContinueKind {
+    var active: ConfiguredRegionState = .inactive
+    XCTAssertNoThrow(try active = node.isActive(in: configuration))
+    XCTAssertEqual(active, .active)
+    return .visitChildren
+  }
+}
+
+class NameCheckingVisitor: ActiveSyntaxAnyVisitor<TestingBuildConfiguration> {
+  /// The set of names we are expected to visit. Any syntax nodes with
+  /// names that aren't here will be rejected, and each of the names listed
+  /// here must occur exactly once.
+  var expectedNames: Set<String>
+
+  init(configuration: TestingBuildConfiguration, expectedNames: Set<String>) {
+    self.expectedNames = expectedNames
+
+    super.init(viewMode: .sourceAccurate, configuration: configuration)
+  }
+
+  deinit {
+    if !expectedNames.isEmpty {
+      XCTFail("No nodes with expected names visited: \(expectedNames)")
+    }
+  }
+
+  func checkName(name: String, node: Syntax) {
+    if !expectedNames.contains(name) {
+      XCTFail("syntax node with unexpected name \(name) found: \(node.debugDescription)")
+    }
+
+    expectedNames.remove(name)
+  }
+
+  open override func visitAny(_ node: Syntax) -> SyntaxVisitorContinueKind {
+    if let identified = node.asProtocol(NamedDeclSyntax.self) {
+      checkName(name: identified.name.text, node: node)
+    } else if let identPattern = node.as(IdentifierPatternSyntax.self) {
+      // FIXME: Should the above be an IdentifiedDeclSyntax?
```

Comment:
Are you proposing to change the name of `NamedDeclSyntax` or `IdentifierPatternSyntax`? If so, could you file an issue for that instead of adding a `FIXME` here, which we probably won’t find again.

Reply:
This is the same sort of issue we're seeing with the lexical name lookup work, where we want some kind of central notion of "syntax node that declares a name"